### PR TITLE
Generic/AbstractClassNamePrefix: improve code coverage

### PR DESCRIPTION
**Pull request by rodrigoprimo**
# Description

This PR improves code coverage for the `Generic.NamingConventions.AbstractClassNamePrefix` sniff and removes an incorrect code comment.

## Related issues/external references

Part of https://github.com/PHPCSStandards/PHP_CodeSniffer/issues/146

## Types of changes

- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_
    - [ ] This change is only breaking for integrators, not for external standards or end-users.
- [ ] Documentation improvement

## PR checklist

- [x] I have checked there is no other PR open for the same change.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md).
- [x] I grant the project the right to include and distribute the code under the BSD-3-Clause license (and I have the right to grant these rights).
- [ ] I have added tests to cover my changes.
- [x] I have verified that the code complies with the projects coding standards.
- [ ] \[Required for new sniffs\] I have added XML documentation for the sniff.

### CODE REVIEW

File: src/Standards/Generic/Tests/NamingConventions/AbstractClassNamePrefixUnitTest.1.inc [hunk 1]
```diff
@@ -56,4 +56,10 @@ class NameAbstractBar {}
 abstract class abstractOkName
 {
 
-}
\ No newline at end of file
+}
+
+final class FinalClassShouldNotTriggerWarning {}
```
Comment (jrfnl) [inline src/Standards/Generic/Tests/NamingConventions/AbstractClassNamePrefixUnitTest.1.inc:61]: What about also adding a test with a `readonly` class ? and an `abstract readonly` class ?

File: src/Standards/Generic/Tests/NamingConventions/AbstractClassNamePrefixUnitTest.1.inc [hunk 2]
```diff
@@ -56,4 +56,10 @@ class NameAbstractBar {}
...
+
+class ChildClassShouldNotTriggerWarning extends AbstractClassName {}
+
+class ClassImplementingInterfaceShouldNotTriggerWarning implements InterfaceName {}
```
Comment (jrfnl) [inline src/Standards/Generic/Tests/NamingConventions/AbstractClassNamePrefixUnitTest.1.inc:65]: These tests aren't testing anything which isn't tested already. The sniff does not look at whether the class extends/implements.

**Review by jrfnl — COMMENTED** [review]
Hi @rodrigoprimo, all good in principle, but can do with some fine-tuning still. See inline comments.

I'm also wondering if, while the test file is being touched anyway, some minor cleanup should be done ?
* `// error` => `// Error.` (multiple times)
* `$var = 'abstract class IncorrectNameButOk';` => `$var = 'abstract class TextStringsAreDisregarded';` (test isn't really testing anything, but I can see why it is there, so let's leave it in, but make it more descriptive)
* `abstract class abstractOkName` => `abstract class abstractOkCaseOfPrefixIsNotEnforced`
* `class NameAbstractBar {}` => `class NotAnAbstractClassSoNoPrefixRequired {}`

Also the `$abstractVar = '';` and `$abstracVar = '';` tests aren't testing anything, so could be removed.

File: src/Standards/Generic/Tests/NamingConventions/AbstractClassNamePrefixUnitTest.1.inc (hunk 1 above)
Comment (rodrigoprimo) [reply src/Standards/Generic/Tests/NamingConventions/AbstractClassNamePrefixUnitTest.1.inc:57]: Done

File: src/Standards/Generic/Tests/NamingConventions/AbstractClassNamePrefixUnitTest.1.inc (hunk 2 above)
Comment (rodrigoprimo) [reply src/Standards/Generic/Tests/NamingConventions/AbstractClassNamePrefixUnitTest.1.inc]: Yes, I added those tests as an extra precaution, but now I checked the `File::getClassProperties()` code, and I can see that it only looks for tokens before `T_CLASS`, so I agree that they are not necessary. I just removed them.

**Comment by rodrigoprimo** [comment]
Thanks for your review, @jrfnl. I replied to the inline comments and implemented the additional cleanup that you suggested. When you get a chance, could you please check this PR again?

**Review by jrfnl — APPROVED** [review]
Thanks @rodrigoprimo ! Looking good.

I'm going to rebase the PR and squash a few commits before merging and I'll make a few more small tweaks to the tests to clean up even further.

**Comment by jrfnl** [comment]
@rodrigoprimo Actually, before I rebase & squash, would you like me to add a separate commit to show the additional tweaks I want to make ?

File: src/Standards/Generic/Tests/NamingConventions/AbstractClassNamePrefixUnitTest.1.inc (hunk 2 above)
Comment (jrfnl) [reply src/Standards/Generic/Tests/NamingConventions/AbstractClassNamePrefixUnitTest.1.inc]: As an alternative, you could have adjusted one or two of the pre-existing tests to have `extends` or `implements`. That would have been fine. But having dedicated tests for it, is redundant.

Comment (jrfnl) [reply src/Standards/Generic/Tests/NamingConventions/AbstractClassNamePrefixUnitTest.1.inc]: Actually, let me just do that now anyway as I added the extra commit(s) now anyway.

**Comment by rodrigoprimo** [comment]
@jrfnl, I just checked the commits you added, and they look good to me. Thanks!

**Comment by jrfnl** [comment]
Rebased & cleaned up the commits. Merging once the build has passed.

**Pull request merged**